pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -610,7 +610,13 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 
 		if len(expr.Args) == 3 {
 			if invokeOptions, ok := expr.Args[2].(*model.ObjectConsExpression); ok {
-				g.Fgen(w, ", new() {\n")
+				// For output form invokes, explicitly use the InvokeOutputOptions class for options
+				// to ensure we pick the correct overload that supports `DependsOn`.
+				if pcl.IsOutputVersionInvokeCall(expr) {
+					g.Fgen(w, ", new InvokeOutputOptions() {\n")
+				} else {
+					g.Fgen(w, ", new() {\n")
+				}
```

Comment:
Actually, I think this change is unnecessary and a minor UX regression for our generated code.

Before calls would look like:

```c#
    var zone = Aws.GetAvailabilityZones.Invoke(new()
    {
        AllAvailabilityZones = true,
    }, new() {
        Provider = explicitProvider,
        Parent = explicitProvider,
        Version = "1.2.3",
        PluginDownloadURL = "https://example.com",
    });
```

...with just `new()`, but now it's uglier because it's explicitly specifying `new InvokeOutputOptions()` rather than letting the compiler infer the type.

But explicitly using `new InvokeOutputOptions()` is unnecessary. `new()` should continue to work, it's just going to pick the new overload that accepts `InvokeOutputOptions` now, which is fine.

Reply:
I thought so initially, but I got an error:

```
tests/testdata/codegen/csharp-invoke-options-pp/dotnet/csharp-invoke-options.cs(32,9): error CS0117: 'InvokeOptions' does not contain a definition for 'DependsOn'
```

Reply:
> but I got an error

I think that's because Aws hasn't been updated to have the new overload that accepts `InvokeOutputOptions` yet, so the compiler is inferring `InvokeOptions` for `new()`.

Reply:
Hmm in that case we need to ship this without the test for program gen, so we can have a release of the AWS SDK with the new code.

Language conformance tests fix this problem in general.

Reply:
Yeah